src/inference/src/dev/core_impl.cpp

```diff
@@ -836,7 +836,12 @@ ov::SoPtr<ov::ICompiledModel> ov::CoreImpl::compile_model(const std::shared_ptr<
         // hint::compiled_blob is set and imported skip compilation
     } else if (cacheManager && device_supports_model_caching(plugin, parsed._config) && !is_proxy_device(plugin)) {
         CacheContent cacheContent{cacheManager, parsed._core_config.get_enable_mmap()};
-        cacheContent.blobId = ov::ModelCache::compute_hash(model, create_compile_config(plugin, parsed._config));
+        cacheContent.blobId = config.count(ov::internal::cache_hash_id.name())
```

Comment:
Can be added local util function to create the cache content ?
The path can be added via ctor.

Why the `config` not parsed config is used?

What if cache_hash_id is wrong then we should not fallback to calculate hash from model path?
and only then fallback compile model path?